### ISSUE TITLE
Gateways disjunction with 2 NICs

### ISSUE THREAD

**Issue by krzkowalczyk** [terraform-providers/terraform-provider-vsphere#153]
Hi there,

I encounter a bug then using 2 NICs, where one of NIC has empty default gateway. Default gateway is populated to both of them in state file, but plan shows that the one which should have a default gateway is actually missing default gateway.

### Terraform Version
Terraform v0.10.2

### Affected Resource(s)
- vsphere_virtual_machine

### Terraform Configuration Files
```hcl
  network_interface {
    label              = "VM Network 964"
    ipv4_address       = "10.128.74.67"
    ipv4_gateway       = "10.128.74.254"
    ipv4_prefix_length = "24"
  }

  network_interface {
    label              = "VM Network 924"
    ipv4_address       = "10.128.34.57"
    ipv4_prefix_length = "24"
  }

```

### Debug Output

State file after first apply action:
```hcl
                            "network_interface.#": "2",
                            "network_interface.0.adapter_type": "",
                            "network_interface.0.ip_address": "",
                            "network_interface.0.ipv4_address": "169.254.172.0",
                            "network_interface.0.ipv4_gateway": "",
                            "network_interface.0.ipv4_prefix_length": "16",
                            "network_interface.0.ipv6_address": "",
                            "network_interface.0.ipv6_gateway": "",
                            "network_interface.0.ipv6_prefix_length": "0",
                            "network_interface.0.key": "4000",
                            "network_interface.0.label": "VM Network 964",
                            "network_interface.0.mac_address": "00:50:56:94:5e:53",
                            "network_interface.0.subnet_mask": "",
                            "network_interface.1.adapter_type": "",
                            "network_interface.1.ip_address": "",
                            "network_interface.1.ipv4_address": "169.254.11.49",
                            "network_interface.1.ipv4_gateway": "",
                            "network_interface.1.ipv4_prefix_length": "16",
                            "network_interface.1.ipv6_address": "",
                            "network_interface.1.ipv6_gateway": "",
                            "network_interface.1.ipv6_prefix_length": "0",
                            "network_interface.1.key": "4001",
                            "network_interface.1.label": "VM Network 924",
                            "network_interface.1.mac_address": "00:50:56:94:10:8b",
                            "network_interface.1.subnet_mask": "",
```
Plan output after first apply
```hcl
  ~ module.nodes.vsphere_virtual_machine.m000p31iis007
      network_interface.0.ipv4_gateway: "" => "10.128.74.254"
```

State file after 2nd apply
```hcl
                            "network_interface.#": "2",
                            "network_interface.0.adapter_type": "",
                            "network_interface.0.ip_address": "",
                            "network_interface.0.ipv4_address": "10.128.74.67",
                            "network_interface.0.ipv4_gateway": "10.128.74.254",
                            "network_interface.0.ipv4_prefix_length": "24",
                            "network_interface.0.ipv6_address": "",
                            "network_interface.0.ipv6_gateway": "",
                            "network_interface.0.ipv6_prefix_length": "0",
                            "network_interface.0.key": "4000",
                            "network_interface.0.label": "VM Network 964",
                            "network_interface.0.mac_address": "00:50:56:94:5e:53",
                            "network_interface.0.subnet_mask": "",
                            "network_interface.1.adapter_type": "",
                            "network_interface.1.ip_address": "",
                            "network_interface.1.ipv4_address": "10.128.34.57",
                            "network_interface.1.ipv4_gateway": "10.128.74.254",
                            "network_interface.1.ipv4_prefix_length": "24",
                            "network_interface.1.ipv6_address": "",
                            "network_interface.1.ipv6_gateway": "",
                            "network_interface.1.ipv6_prefix_length": "0",
                            "network_interface.1.key": "4001",
                            "network_interface.1.label": "VM Network 924",
                            "network_interface.1.mac_address": "00:50:56:94:10:8b",
                            "network_interface.1.subnet_mask": "",
```

### Expected Behavior
Terraform sets up network interfaces according to definition in .tf file. One interface has a default gateway and the second one doesn't have it.
There are no differences after applying resources.

### Actual Behavior
Terraform creates node and applies network interfaces configuration. After applying, generating plan shows difference in gateway configuration, gateway is missing.
```hcl
  ~ module.nodes.vsphere_virtual_machine.m000p31iis007
      network_interface.0.ipv4_gateway: "" => "10.128.74.254"
```
Reapplying resource ends with success, but the difference is still present.

Terraform says that NIC 0 doesn't have gateway, which is not true. On the OS level gateway is properly assigned. 
In addition in state file it's stated that NIC1 has default gateway, and the address places here is the address from default gateway from NIC0.

I will test this later on with 2+ NICs

This issue was not present in v0.1.0 of provider.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. Create a .tf file with 2 NIC, where on one NIC default gateway is empty or not provided.
2. `terraform apply`
3. `terraform plan`

### Important Factoids
In first state file NICs have apipa addresses, but i believe this is related to terraform not waiting for end of vsphere customization.

**Comment by krzkowalczyk** [comment]
With 4 NIC's, where default gateway is assigned only to first NIC. 
In state file, the last NIC in array has assigned default gateway together with first one.
Plan shows the first NIC is missing a gateway

>   ~ module.nodes.vsphere_virtual_machine.m000p31iis012
>       network_interface.0.ipv4_gateway: "" => "10.128.74.254"

**Comment by vancluever** [comment]
Hey @krzkowalczyk, sorry for the delay in replying to this!

I think I may have fixed this as a side effect of #180 as it seems like it could possibly be due to the same code.

If it is not much trouble, can you try with a [custom built provider binary](https://github.com/terraform-providers/terraform-provider-vsphere#building-the-provider) against the branch referenced in that PR to see if that fixes it for you?

Thanks!

**Comment by krzkowalczyk** [comment]
Hey @vancluever,
after short test with patched provider it seems the problem does not occur any more.
Thanks for your support !

**Comment by vancluever** [comment]
Glad it fixed the issue! Going to close this now as #180 won't close out this one when it's merged... thanks! :+1:
